### PR TITLE
Closes #492: Support 1.1.1 contracts

### LINKED ISSUE #492: Prepare app for contract upgrade
The new version of the contracts have been audited and should be used by the apps by end of the year.

For more info see https://github.com/gnosis/safe/issues/262

### TODOs:
- [ ] Add migration transaction for existing Safes (this needs to add the fallback manager)
  - https://github.com/gnosis/safe-contracts/releases/tag/v1.1.0
- [ ] Use new creation endpoint
  - gnosis/safe-relay-service#160

### PR DESCRIPTION
**Pull request by rmeissner**
Closes #492.

Changes proposed in this pull request:
- Add new master copy
- Change update flow to check for new master copy

@gnosis/mobile-devs

### CODE REVIEW

File: app/src/main/java/pm/gnosis/heimdall/data/repositories/impls/DefaultTransactionInfoRepository.kt [hunk 1]
```diff
@@ -147,9 +149,40 @@ class DefaultTransactionInfoRepository @Inject constructor(
     }
 
     private fun processMultiSend(transaction: SafeTransaction, multiSend: TransactionData.MultiSend) =
-        parseReplaceRecoveryPhrase(transaction, multiSend)
+        parseUpgradeV110(multiSend)
+            ?: parseReplaceRecoveryPhrase(transaction, multiSend)
             ?: multiSend
 
+    private fun parseUpgradeV110(multiSend: TransactionData.MultiSend): TransactionData? {
```
Comment (sche) [inline app/src/main/java/pm/gnosis/heimdall/data/repositories/impls/DefaultTransactionInfoRepository.kt]: should it be V111 everywhere?

File: app/build.gradle [hunk 2]
```diff
@@ -21,15 +21,14 @@ android {
...
         // Contracts
-        buildConfigField javaTypes.STRING, "SAFE_MASTER_COPY_0_0_2", asString(getKey("SAFE_MASTER_COPY_0_0_2", "0xAC6072986E985aaBE7804695EC2d8970Cf7541A2"))
-        // Version 0.0.2-alpha (Mainnet)
-        buildConfigField javaTypes.STRING, "SAFE_MASTER_COPY_0_1_0", asString(getKey("SAFE_MASTER_COPY_0_1_0", "0x8942595A2dC5181Df0465AF0D7be08c8f23C93af"))
-        // Version 0.1.0 (All networks)
-        buildConfigField javaTypes.STRING, "SAFE_MASTER_COPY_1_0_0", asString(getKey("SAFE_MASTER_COPY_1_0_0", "0xb6029EA3B2c51D09a50B53CA8012FeEB05bDa35A"))
-        // Version 1.0.0 (All networks)
-        buildConfigField javaTypes.STRING, "PROXY_FACTORY_ADDRESS", asString(getKey("PROXY_FACTORY_ADDRESS", "0x12302fE9c02ff50939BaAaaf415fc226C078613C"))
+        buildConfigField javaTypes.STRING, "SAFE_MASTER_COPY_0_0_2", asString(getKey("SAFE_MASTER_COPY_0_0_2", "0xAC6072986E985aaBE7804695EC2d8970Cf7541A2")) // Version 0.0.2-alpha (Mainnet)
+        buildConfigField javaTypes.STRING, "SAFE_MASTER_COPY_0_1_0", asString(getKey("SAFE_MASTER_COPY_0_1_0", "0x8942595A2dC5181Df0465AF0D7be08c8f23C93af")) // Version 0.1.0 (All networks)
```
Comment (elgatovital) [inline app/build.gradle]: How is it possible to deploy contract under same address on different networks?

File: app/src/main/java/pm/gnosis/heimdall/data/remote/RelayServiceApi.kt [hunk 3]
```diff
@@ -14,24 +14,24 @@ interface RelayServiceApi {
     @POST("v1/safes/{address}/transactions/estimate/")
     fun estimate(@Path("address") address: String, @Body params: EstimateParams): Single<RelayEstimate>
 
-    @POST("v2/safes/")
-    fun safeCreation(@Body params: RelaySafeCreationParams): Single<RelaySafeCreation>
+    @GET("v1/tokens/?limit=1000&ordering=relevance,name&gas=true")
```
Comment (DmitryBespalov) [inline app/src/main/java/pm/gnosis/heimdall/data/remote/RelayServiceApi.kt]: you can as well change the limit to 3000 (not related to this pr, though) 😉

Comment (rmeissner) [reply app/src/main/java/pm/gnosis/heimdall/data/remote/RelayServiceApi.kt]: yes, this is not part of this PR ;)

**Pull request merged**